server/src/Q3Server/AuthParameterChecker.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.Threading.Tasks;
+using Microsoft.Owin;
+
+namespace Q3Server
+{
+    public class AuthParameterChecker : OwinMiddleware
+    {
+        public AuthParameterChecker(OwinMiddleware next) : base(next)
+        {
+        }
+
+        public override async Task Invoke(IOwinContext context)
+        {
+            if (HasCorrectParameterCombination(context.Request))
+            {
+                await Next.Invoke(context);
+            }
+            else
+            {
+                context.Response.StatusCode = 400;

```

Comment:
Or indeed, should we instead just be returning a new HttpUnauthorisedResult? https://msdn.microsoft.com/en-us/library/system.web.mvc.httpunauthorizedresult(v=vs.118).aspx

I guess this isn't the controller so it's less likely